test/CMakeLists.txt

```
@@ -102,10 +109,14 @@ foreach(file ${sources})
     endforeach(test_raw)
 endforeach(file)
 
+#Save allTests into a variable in include file
+configure_file(tools/libtesteth/cmakeAllTests.h.in tools/libtesteth/cmakeAllTests.h)
```

```suggestion
configure_file(tools/libtesteth/AllTestNames.h.in ${PROJECT_BINARY_DIR}/AllTestNames.h)
```

Better name suggestion: `AllTestNames.h`
By generating the file to the root binary dir you don't have to add additional include dir because the root one is already included.